source/command-logging-and-monitoring/command-logging-and-monitoring.rst

```diff
@@ -340,42 +359,166 @@ See the `Load Balancer Specification <../load-balancers/load-balancers.rst#event
+
+The types used in the structured message definitions below are demonstrative, and drivers MAY use similar types instead so long as the information
+is present (e.g. a double instead of an integer, or a string instead of an integer if the structured logging framework does not support numeric types.)
+
+Drivers MUST not emit command log messages for commands issued as part of the handshake with the server, or heartbeat commands issued by server monitors. 
+
+Common Fields
+-------------
+The following key-value pairs MUST be included in all command messages:
 
-A Ruby subscriber to a query series and how it could handle it with respect to logging.
+.. list-table::
+   :header-rows: 1
+   :widths: 1 1 1
+
+   * - Key
+     - Suggested Type
+     - Value
+
+   * - commandName
+     - String
+     - The command name.
+
+   * - requestId
+     - Int
+     - The driver-generated request ID.
+  
+   * - operationId
+     - Int
+     - The driver-generated operation ID. Optional; only present if the driver generated operation IDs and this command has one. 
+
+   * - driverConnectionId
+     - Int
+     - The driver's ID for the connection used for the command. Note this is NOT the same as ``CommandStartedEvent.connectionId`` defined above,
+       but refers to the `connectionId` defined in the  `connection monitoring and pooling specification <../connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst>`_.
+       Unlike ``CommandStartedEvent.connectionId`` this field MUST NOT contain the host/port; that information MUST be in the following fields,
+       ``serverHost`` and ``serverPort``. This field is optional for drivers that do not implement CMAP if they do have an equivalent concept of
+       a connection ID.
+
+   * - serverHost
+     - String
+     - The hostname or IP address for the server the command is being run on.
+
+   * - serverPort
+     - Int
+     - The port for the server the command is being run on. Optional; only present if a port was specified.
+
+   * - serverConnectionId
+     - Int
+     - The server's ID for the connection used for the command. Optional; only present for server versions 4.2+.
+
+   * - serviceId
+     - String
+     - The hex string representation of the service ID for the command. Optional; only present when the driver is in load balancer mode.
+
+Command Started Message
+-----------------------
+In addition to the common fields, command started messages MUST contain the following key-value pairs:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 1 1 1
+
+   * - Key
+     - Suggested Type
+     - Value
+
+   * - message
+     - String
+     - "Command started"
+
+   * - command
+     - String
+     - Relaxed extJSON representation of the command. This document MUST be truncated appropriately according to rules defined in the 
+       `logging specification <../logging/logging.rst>`_, and MUST be replaced with an empty document "{ }" if the command is considered sensitive.
```

Comment:
For your consideration, you could add a link to the truncation section #truncation-of-large-documents.

Reply:
good thought to add a link. that section covers truncation rationale, but I am going to add one to the section on truncation rules: Configurable Max Document Length.